/*
 * SPDX-License-Identifier: EUPL-1.2 OR LicenseRef-commercial
 *
 * Copyright (c) 2012-2026 mgm technology partners GmbH
 *
 * Dual License
 * ------------
 * This source file is part of the mgm A12 Platform and available under
 * a choice of two different licenses:
 *
 * 1. Open-Source License - EUPL v1.2
 *    You may redistribute and/or modify this file under the terms of the
 *    European Union Public License, version 1.2 - see https://eupl.eu/.
 *
 * 2. Commercial License
 *    Alternatively, you may obtain a commercial license from
 *    mgm technology partners GmbH, that permits use of this software
 *    under different terms (including support and maintenance services).
 *
 *    Please contact a12-license@mgm-tp.com for more information.
 *
 * You must select and comply with exactly one of the above license options.
 *
 * Warranty Disclaimer (applies to either option)
 * ----------------------------------------------
 * THIS SOFTWARE IS PROVIDED "AS IS" AND WITHOUT WARRANTY OF ANY KIND,
 * WHETHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
 * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 * NON-INFRINGEMENT, EXCEPT WHERE SUCH DISCLAIMERS ARE HELD TO BE
 * LEGALLY INVALID. SEE THE RESPECTIVE LICENSE TEXT FOR DETAILS.
 */
package com.mgmtp.a12.print.setting.internal.model.impl;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.mgmtp.a12.print.setting.internal.model.Attachment;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Value;
import lombok.experimental.SuperBuilder;

import java.io.Serializable;
import java.util.Optional;

@Value
@NoArgsConstructor(force = true, access = AccessLevel.PROTECTED)
@SuperBuilder
public class AttachmentDto implements Attachment {
	@JsonProperty(value = "internal_filename", required = true)
	String internalFilename;
	@JsonProperty(value = "mime_type", required = true)
	String mimeType;
	@JsonProperty(value = "content", required = true)
	String content;
	@JsonProperty(value = "size", required = true)
	Integer size;
	@JsonProperty("original_filename")
	String originalFilename;
	@JsonProperty("category")
	String category;
	@JsonProperty("description")
	String description;
	@JsonProperty("attachment_id")
	String attachmentId;

	public Optional<String> getOriginalFilename() {
		return Optional.ofNullable(originalFilename);
	}
	public Optional<String> getCategory() {
		return Optional.ofNullable(category);
	}
	public Optional<String> getDescription() {
		return Optional.ofNullable(description);
	}
	public Optional<String> getAttachmentId() {
		return Optional.ofNullable(attachmentId);
	}
}
